packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -189,16 +189,18 @@ export const treeNodeApi: TreeNodeApi = {
 			T
 		>;
 	},
-	shortId(node: TreeNode): number | undefined {
+	shortId(node: TreeNode): number | string | undefined {
 		const flexNode = getFlexNode(node);
 		for (const field of flexNode.boxedIterator()) {
 			if (field.schema.kind === FieldKinds.identifier) {
 				const identifier = field.boxedAt(0);
 				assert(identifier !== undefined, 0x927 /* The identifier must exist */);
-
-				return extractFromOpaque(
-					identifier.context.nodeKeys.localize(identifier.value as StableNodeKey),
+				const localNodeKey = identifier.context.nodeKeys.tryLocalize(
+					identifier.value as StableNodeKey,
```

Comment:
Remove this cast - cast to string instead. We _know_ that value is a string, so that's a fine cast. We do _not_ know that it is a `StableNodeKey`, so that's _not_ a fine cast!

You're also casting to string below - you can just cast once above: `const identifier = field.boxedAt(0) as string`.

Reply:
I couldn't access `isStableId` from `treeApi.ts`, so I added a type guard function `isStableNodeKey` to `nodeKeys` and `nodeKeyManager` so that it could be used here